### ISSUE TITLE
Kotlin support

### ISSUE THREAD

**Issue by vRallev** [uber/okbuck#323]
I read that Kotlin is supported by Buck now. In the Fragment podcast it was mentioned that the Kotlin support is WIP. The [features wiki](https://github.com/uber/okbuck/wiki/Features) states that it's planned. But I couldn't find a ticket to track progress. Is there maybe an experimental snapshot build already?

**Comment by kageiit** [comment]
There is no snapshot yet. Thanks for opening the issue. We are waiting on better kotlin support in buck (its quite basic right now) and plan to open the corresponding support PR in okbuck right after

**Comment by kageiit** [comment]
Specifically waiting for https://github.com/brettwooldridge/buck/tree/kotlin-compiler to merge back into usptream. cc @brettwooldridge

**Comment by brettwooldridge** [comment]
My current estimate is ~2 weeks before a pull request can be submitted...

**Comment by vRallev** [comment]
Thanks, that sounds great. I'm also glad to hear that there's already a lot progress.

**Comment by traviswestbrook** [comment]
Just wanted to give this a bump and see if there was anything that could be done to help out :)

**Comment by kageiit** [comment]
Nothing much needed from the okbuck side. compiling mixed sources and annotation processor support is currently missing in buck. Creating the bridge between gradle to buck via okbuck once those features are available will not be much work

**Comment by kageiit** [comment]
> Just wanted to give this a bump and see if there was anything that could be done to help out :)

You can help with the conversation in
https://github.com/facebook/buck/issues/1037
https://github.com/facebook/buck/issues/956

and see if you can help out with the buck codebase

**Comment by seanabraham** [comment]
@brettwooldridge any updates on your end? Is this the latest on in-memory support? https://github.com/brettwooldridge/buck/commit/61fddf0ead19695b5d0ddafec96c34d98b698621

**Comment by kageiit** [comment]
> My current estimate is ~2 weeks before a pull request can be submitted...

@brettwooldridge Any updates on the kotlin PR? Please let us know if there is any way we can move forward the kotlin in memory compiler support at the minimum

**Comment by runningcode** [comment]
We should close this issue in favor of issue #400 .
In memory compiler support has been merged.
Buck now supports kotlin, kotlin android module, and kotlin and java mixed sources.

**Comment by kageiit** [comment]
Duplicate of #400
